src/qt/bitcoinamountlabel.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qt/bitcoinamountlabel.h>
+
+#include <QRegularExpression>
+
+#include <cassert>
+
+BitcoinAmountLabel::BitcoinAmountLabel(QWidget* parent)
+    : QLabel(parent)
+{
+}
+
+void BitcoinAmountLabel::setText(const QString& newText)
+{
+    if (cache.isEmpty()) {
+        QLabel::setText(newText);
+    } else {
+        cache = newText;
+    }
+}
+
+void BitcoinAmountLabel::changePrivacyMode(bool privacy)
+{
+    static const QString mask{'#'};
+    if (privacy && cache.isEmpty()) {
+        cache = text();
+        QStringList parts = text().split('.');
+        assert(parts.size() == 2);
+        QString integer_part = parts.first();
+        integer_part = QString(integer_part.size() - 1, ' ') + mask;
+        QString fractional_part = parts.last();
+        fractional_part = fractional_part.replace(QRegularExpression("\\d"), mask);
+        QLabel::setText(integer_part + '.' + fractional_part);
```

Comment:
Just looked at how Wasabi did this and they just replace any amounts, addresses, labels everywhere with a fixed number of #'s. IMO "#########" even looks better than "#.########" and it's less lines of code, all this block could be just replaced with
```
cache = text();
QLabel::setText("#########");`
```

Reply:
Unfortunately, this will cause the UI to reflow the columns on every toggle (see @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/16432#discussion_r305807357)).